src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -356,74 +367,95 @@ IndexerConnector::IndexerConnector(
-                // If the element should not be indexed, only delete it from the sync database.
-                const bool noIndex = parsedData.contains("no-index") ? parsedData.at("no-index").get<bool>() : false;
-
-                if (parsedData.at("operation").get_ref<const std::string&>().compare("DELETED") == 0)
+                if (!noIndex)
                 {
-                    if (!noIndex)
-                    {
-                        builderBulkDelete(bulkData, id, m_indexName);
-                    }
-                    m_db->delete_(id);
+                    builderBulkDelete(bulkData, id, m_indexName);
                 }
-                else
+                m_db->delete_(id);
+            }
+            else
+            {
+                const auto dataString = parsedData.at("data").dump();
+                if (!noIndex)
                 {
-                    const auto dataString = parsedData.at("data").dump();
-                    if (!noIndex)
-                    {
-                        builderBulkIndex(bulkData, id, m_indexName, dataString);
-                    }
-                    m_db->put(id, dataString);
+                    builderBulkIndex(bulkData, id, m_indexName, dataString);
                 }
+                m_db->put(id, dataString);
             }
+        }
 
-            if (!bulkData.empty())
-            {
-                // Process data.
-                HTTPRequest::instance().post(
-                    HttpURL(url),
-                    bulkData,
-                    [](const std::string& response) { logDebug2(IC_NAME, "Response: %s", response.c_str()); },
-                    [](const std::string& error, const long statusCode)
-                    {
-                        logError(IC_NAME, "%s, status code: %ld", error.c_str(), statusCode);
-                        throw std::runtime_error(error);
-                    },
-                    "",
-                    DEFAULT_HEADERS,
-                    secureCommunication);
-            }
-        },
-        DATABASE_BASE_PATH + m_indexName,
-        ELEMENTS_PER_BULK);
+        if (!bulkData.empty())
+        {
+            // Process data.
+            HTTPRequest::instance().post(
+                HttpURL(url),
+                bulkData,
+                [](const std::string& response) { logDebug2(IC_NAME, "Response: %s", response.c_str()); },
+                [](const std::string& error, const long statusCode)
+                {
+                    logError(IC_NAME, "%s, status code: %ld", error.c_str(), statusCode);
+                    throw std::runtime_error(error);
+                },
+                "",
+                DEFAULT_HEADERS,
+                secureCommunication);
+        }
+    };
+
+    try
+    {
+        m_dispatcher =
+            std::make_unique<ThreadDispatchQueue>(functor, DATABASE_BASE_PATH + m_indexName, ELEMENTS_PER_BULK);
+    }
+    catch (std::system_error& e)
```

Comment:
We can consider declaring the exception object as `const` in the catch clause.

```suggestion
    catch (const std::system_error& e)
```

Reply:
You are right! Thank you.